### ISSUE TITLE
LightController related issue when trying to add enemies 

### ISSUE THREAD

**Issue by ronnie-dumesh** [BryanGraeser/fallen-flame#28]
This can be encountered in the add-enemies branch 

I replaced the JSON file for enemies from "enemies" : [] to 
`
  "enemies":          [
    {
      "pos":        	[ 13.0, 6.0],
      "radius":      	0.4,
      "bodytype":    	"dynamic",
      "density":        1.0,
      "friction":       0.0,
      "restitution":    0.0,
      "force": 		 80.0,
      "damping": 		 10.0,
      "maxspeed": 	  10.0,
      "startframe":       0,
      "walklimit":		4,
      "texture":     	"player",
      "debugcolor":  	"white",
      "debugopacity":	192
    }
  ],`

It more or less mirrors the JSON code for a player except I changed the position. After some debugging, I am now stuck due to an issue with LightController. Upon starting the game, I received this error. 

Exception in thread "LWJGL Application" java.util.ConcurrentModificationException
	at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1561)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	**at com.fallenflame.game.LightController.updateLights(LightController.java:182)**
	at com.fallenflame.game.LevelController.update(LevelController.java:410)
	at com.fallenflame.game.GameEngine.update(GameEngine.java:283)
	at com.fallenflame.game.GameEngine.render(GameEngine.java:337)
	at com.badlogic.gdx.Game.render(Game.java:46)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.mainLoop(LwjglApplication.java:233)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:128)

Commenting out the line of code caused the game to freeze and eventually crash. I believe this may be an issue with AI-Controller/LevelController and I am looking into that. However, it is possible it has to do with the absence of this code

**Comment by ronnie-dumesh** [comment]
Currently, line 182 of LightController is commented out. It must not be to reproduce the error

**Comment by BryanGraeser** [comment]
Yeah so I looked at the code and the issue is for this code:
```
enemyLights.keySet().stream().filter(i -> !enemies.contains(i) || !i.getActivated()).forEach(i -> {
            PointSource f = enemyLights.get(i);
            f.setActive(false);
            f.dispose();
            enemyLights.remove(i);
        });
```
You can't stream/filter over enemyLights.keySet() while simultaneously changing the size of enemyLights.keySet() (via `enemyLights.remove(i);`).

I can rewrite this real quick so this is not the case.

**Comment by BryanGraeser** [comment]
Solved in this PR: #34 , however PR: #33 needs to be reviewed first

**Comment by ronnie-dumesh** [comment]
Resolved, both PRs merged
